// splinterlands-types/lib/api.js
"use strict";
/** Types for the Splinterlands API */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardAttackType = exports.Ability = exports.CardType = exports.CardColor = void 0;
var CardColor;
(function (CardColor) {
    CardColor["BLACK"] = "Black";
    CardColor["BLUE"] = "Blue";
    CardColor["GOLD"] = "Gold";
    CardColor["GRAY"] = "Gray";
    CardColor["GREEN"] = "Green";
    CardColor["RED"] = "Red";
    CardColor["WHITE"] = "White";
})(CardColor = exports.CardColor || (exports.CardColor = {}));
var CardType;
(function (CardType) {
    CardType["SUMMONER"] = "Summoner";
    CardType["MONSTER"] = "Monster";
})(CardType = exports.CardType || (exports.CardType = {}));
/*
 * How to generate:
 * https://api2.splinterlands.com/settings
 * var abilityToEnumTS = function(ability) {
 *   var enumName = ability.name.toUpperCase().split(' ').join('_');
 *   return `${enumName} = '${ability.name}',`;
 * }
 * window.abilities.forEach((ability) => str+= abilityToEnumTS(ability));
 * copy(str);
 */
var Ability;
(function (Ability) {
    Ability["AFFLICTION"] = "Affliction";
    Ability["AMPLIFY"] = "Amplify";
    Ability["BACKFIRE"] = "Backfire";
    Ability["BLAST"] = "Blast";
    Ability["BLIND"] = "Blind";
    Ability["BLOODLUST"] = "Bloodlust";
    Ability["CAMOUFLAGE"] = "Camouflage";
    Ability["CLEANSE"] = "Cleanse";
    Ability["CLOSE_RANGE"] = "Close Range";
    Ability["CRIPPLE"] = "Cripple";
    Ability["DEATHBLOW"] = "Deathblow";
    Ability["DEMORALIZE"] = "Demoralize";
    Ability["DISPEL"] = "Dispel";
    Ability["DIVINE_SHIELD"] = "Divine Shield";
    Ability["DODGE"] = "Dodge";
    Ability["DOUBLE_STRIKE"] = "Double Strike";
    Ability["ENRAGE"] = "Enrage";
    Ability["FLYING"] = "Flying";
    Ability["FORCEFIELD"] = "Forcefield";
    Ability["GIANT_KILLER"] = "Giant Killer";
    Ability["HALVING"] = "Halving";
    Ability["HEADWINDS"] = "Headwinds";
    Ability["HEAL"] = "Heal";
    Ability["IMMUNITY"] = "Immunity";
    Ability["INSPIRE"] = "Inspire";
    Ability["KNOCK_OUT"] = "Knock Out";
    Ability["LAST_STAND"] = "Last Stand";
    Ability["LIFE_LEECH"] = "Life Leech";
    Ability["MAGIC_REFLECT"] = "Magic Reflect";
    Ability["OPPORTUNITY"] = "Opportunity";
    Ability["OPPRESS"] = "Oppress";
    Ability["PHASE"] = "Phase";
    Ability["PIERCING"] = "Piercing";
    Ability["POISON"] = "Poison";
    Ability["PROTECT"] = "Protect";
    Ability["REACH"] = "Reach";
    Ability["RECHARGE"] = "Recharge";
    Ability["REDEMPTION"] = "Redemption";
    Ability["REFLECTION_SHIELD"] = "Reflection Shield";
    Ability["REPAIR"] = "Repair";
    Ability["RESURRECT"] = "Resurrect";
    Ability["RETALIATE"] = "Retaliate";
    Ability["RETURN_FIRE"] = "Return Fire";
    Ability["RUST"] = "Rust";
    Ability["SCATTERSHOT"] = "Scattershot";
    Ability["SCAVENGER"] = "Scavenger";
    Ability["SHATTER"] = "Shatter";
    Ability["SHIELD"] = "Shield";
    Ability["SILENCE"] = "Silence";
    Ability["SLOW"] = "Slow";
    Ability["SNARE"] = "Snare";
    Ability["SNEAK"] = "Sneak";
    Ability["SNIPE"] = "Snipe";
    Ability["STRENGTHEN"] = "Strengthen";
    Ability["STUN"] = "Stun";
    Ability["SWIFTNESS"] = "Swiftness";
    Ability["TANK_HEAL"] = "Tank Heal";
    Ability["TAUNT"] = "Taunt";
    Ability["THORNS"] = "Thorns";
    Ability["TRAMPLE"] = "Trample";
    Ability["TRIAGE"] = "Triage";
    Ability["TRUE_STRIKE"] = "True Strike";
    Ability["VOID"] = "Void";
    Ability["VOID_ARMOR"] = "Void Armor";
    Ability["WEAKEN"] = "Weaken";
    Ability["FURY"] = "Fury";
})(Ability = exports.Ability || (exports.Ability = {}));
var CardAttackType;
(function (CardAttackType) {
    CardAttackType["MELEE"] = "melee";
    CardAttackType["RANGED"] = "ranged";
    CardAttackType["MAGIC"] = "magic";
})(CardAttackType = exports.CardAttackType || (exports.CardAttackType = {}));
